niceman/resource/ssh.py

```diff
@@ -133,34 +135,36 @@ def _execute_command(self, command, env=None, cwd=None):
 
         # If a command fails, a CommandError exception will be thrown.
         escaped_command = ' '.join(quote(s) for s in command)
-        for i, line in enumerate(self.ssh.execute(escaped_command)):
-            lgr.debug("exec#%i: %s", i, line.rstrip())
+        out = []
+        try:
+            for i, line in enumerate(self.ssh.execute(escaped_command)):
+                out.append(line.rstrip())
+                lgr.debug("exec#%i: %s", i, line.rstrip())
+        except RemoteCommandFailed as e:
+            raise CommandError(cmd=escaped_command, msg=e.msg)
+
+        return (out, None)
```

Comment:
same as with docker:  what is happening to stderr?
generally it is passed back:
```shell
$> ssh smaug.datalad.org "bash -c 'echo 123 >&2'" > /dev/null
123
```